### PR TITLE
Warn or fail a PR if increases total repo size in bytes by more than thresholds

### PR DESCRIPTION
**Pull request by jamesrb1**
We want to keep our main branch slim to ensure it is quick to clone, including by package managers. The changes here will warn/block PRs where the file size exceeds thresholds, currently set to warn at 100kb and fail at 250kb (up for discussion!).

This is an example of a PR that generates a warning:

image

### CODE REVIEW

**Review by vegaro — APPROVED** [review]
Love you added both a warning and an error!

**Review by tonidero — APPROVED** [review]
Nice idea! I'm wondering if we might want a way to bypass this check, maybe with a custom label or something like that? I'm thinking in case we ever need to add an image or something... But this is good enough for now!

File: Dangerfile [hunk 1]
```diff
@@ -27,5 +29,63 @@ def fail_if_no_supported_label_found
   end
 end
 
+#### PR SIZE INCREASE CHECK
+
+# Thresholds
+WARN_SIZE_INCREASE = 102400  # 100kb
+FAIL_SIZE_INCREASE = 256000  # 250kb
```
Comment (aboedo) [inline Dangerfile:36]: the limits feel a bit aggressive but that might be a good thing, we can fine-tune them later if they're limiting us in any way.

**Review by aboedo — APPROVED** [review]
awesome!

**Comment by jamesrb1** [comment]
I will follow up with another PR adding the label bypass.

**Pull request merged**